dll/win32/msgina/CMakeLists.txt

```diff
@@ -10,28 +10,27 @@ include_directories(
 
 spec2def(msgina.dll msgina.spec)
 
-list(APPEND C_SOURCE
+list(APPEND SOURCE
     gui.c
     lsa.c
     msgina.c
     shutdown.c
     stubs.c
-    tui.c
-    msgina.h)
+    tui.c)
 
-list(APPEND CPP_SOURCE
-    dimmedwindow.cpp)
+list(APPEND PCH_SKIP_SOURCE
+    dimmedwindow.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/msgina_stubs.c)
 
 add_library(msgina MODULE
-    ${C_SOURCE}
-    ${CPP_SOURCE}
+    ${SOURCE}
+    ${PCH_SKIP_SOURCE}
     msgina.rc
-    ${CMAKE_CURRENT_BINARY_DIR}/msgina_stubs.c
     ${CMAKE_CURRENT_BINARY_DIR}/msgina.def)
 
 set_module_type(msgina win32dll UNICODE)
 target_link_libraries(msgina wine uuid ${PSEH_LIB})
 add_delay_importlibs(msgina secur32)
 add_importlibs(msgina advapi32 user32 gdi32 powrprof userenv msvcrt kernel32 ntdll)
-add_pch(msgina msgina.h CPP_SOURCE)
+add_pch(msgina msgina.h "${PCH_SKIP_SOURCE}")
```

Comment:
This is interesting -- the PCH was previously only applied to the .cpp file. I suppose that was a mistake?

Reply:
Yeah, I had the exact same reaction! I deemed it an oversight because we had one c++ file and many c files so it makes sense to leverage PCH for the c ones :)

Reply:
iirc, sharing PCH between c and c++ caused build problems.